etc/run.sh

```diff
@@ -0,0 +1,243 @@
+        exit 1
+    fi
+    _exec_all
+}
+
+_op_jupyterhub() {
+    # POSIT: versions same in container-beamsim-jupyter/build.sh
+    # Order is important: jupyterlab-server should be last so it isn't
+    # overwritten with a newer version.
+    declare p=$(pip freeze)
+    declare -a i=()
+    declare f
+    for f in \
+        jupyterhub==1.4.2 \
+        jupyterlab==3.1.14  \
+        notebook==6.5.4 \
+        jupyterlab_server==2.8.2\
+        ; do
+        if ! [[ $p =~ $f ]]; then
+            i+=( $f )
+        fi
+    done
+    if (( ${#i[@]} > 0 )); then
+        pip install "${i[@]}"
+    fi
+    if ! type configurable-http-proxy &> /dev/null; then
+        npm install --global configurable-http-proxy
+    fi
+    _env_moderate jupyterhublogin
+    sirepo service tornado &
+    sirepo service nginx-proxy &
+    sirepo job_supervisor &
+    sirepo service jupyterhub &
+    declare -a x=( $(jobs -p) )
+    # this doesn't kill uwsgi for some reason; TERM is better than KILL
+    trap "kill ${x[*]}" EXIT
+    wait -n
+}
+
+_op_ldap() {
+    if ! systemctl is-active slapd &> /dev/null; then
+       _msg "setting up ldap/slapd"
+       cd "$(dirname "$0")"
+       bash setup-ldap.sh
+       cd -
+    fi
+    export SIREPO_AUTH_METHODS=guest:ldap
+    _msg 'To test:
+
+Login as vagrant@radiasoft.net/vagrant
+'
+    _exec_all
+}
+
+
+_op_mail() {
+    _setup_smtp
+    _exec_all
+}
+
+_op_moderate() {
+    _env_moderate srw
+    _exec_all
+}
+
+_op_no_smtp_mail() {
+    # POSIT: same as sirepo.smtp.DEV_SMTP_SERVER
+    export SIREPO_SMTP_SERVER=dev
+    _env_mail_common
+    _exec_all
+}
+
+_op_react() {
+    export SIREPO_FEATURE_CONFIG_UI_REACT=1
+    # Only one auth method allowed
+    export SIREPO_AUTH_METHODS=email
+    # Websockets don't work
+    export SIREPO_FEATURE_CONFIG_UI_WEBSOCKET=0
+    _op_mail
+}
+
+_op_react_build() {
+    if [[ ! ${run_react_build_no_compile:-} ]]; then
+        cd "$(dirname "$0")"/../react
+        rm -rf build
+        npm run-script build
+        (
+            # These aren't likely to fail so run in subshell
+            cd ..
+            rm -f sirepo/package_data/static/react
+            ln -s ../../../react/build sirepo/package_data/static/react
+        )
+    fi
+    export SIREPO_PKCLI_SERVICE_REACT_PORT=
+    export SIREPO_SERVER_REACT_SERVER=build
+    _op_react
+}
+
+_op_server_status() {
+    declare u=$(cd "$(dirname "$0")"/../run/user && ls -d ???????? 2>/dev/null | head -1)
+    if [[ ! $u ]]; then
+        _err 'Start the server first to create a user and then server_status can work'
+    fi
+    export SIREPO_AUTH_BASIC_PASSWORD=password
+    export SIREPO_AUTH_BASIC_UID=$u
+    export SIREPO_AUTH_METHODS=guest:basic
+    export SIREPO_FEATURE_CONFIG_API_MODULES=status
+    _msg "To test:
+
+curl -u '$u:$SIREPO_AUTH_BASIC_PASSWORD' http://localhost:8000/server-status
+"
+    _exec_all
+}
+
+_op_test_mail() {
```

Comment:
Haven't looked too deep into it but this fails for me locally when ~/mail already exists, and probably unrelated, tests/auth/email1_test.py is failing ci

Reply:
7cac2f975. email1_test failure is described in #6383.